numba/tests/test_firstlinefinder.py

```diff
@@ -79,6 +80,32 @@ def test_string_eval(self):
         # Cannot determine first line of string evaled functions
         self.assertIsNone(first_def_line)
 
+    def test_string_eval_linecache(self):
+        source = """def foo():
+            pass
+        """
```

Comment:
i am testing it with:
```python
        source = """if True:
        # indent causes IndentationError in ast.parse

        def decor(f):
            return f

        @decor 
        @decor
        @decor
        def foo():
            pass
        """
```
indentation is not handled. As `ast.parse()` is seeing:

```python
        @decor  # nested decorators work fine
        @decor
        @decor
        def foo():
            pass
```

Reply:
Thanks - I've added some tests now that do something similar.